Add Catalog-Aware Kerberos Authentication for Multi-Domain Environments (#25822)
prestodb/presto: presto-docs/src/main/sphinx/connector/hive-catalog-aware-kerberos.rst

> ==================================

Reviewer: ```suggestion
=====================================
```
The formatting line must be exactly the same number of characters as the line being formatted, or warnings like the following appear in the doc build logs: 

```
/Users/steveburnett/Documents/GitHub/presto/presto-docs/src/main/sphinx/connector/hive-catalog-aware-kerberos.rst:1: WARNING: Title overline too short.

==================================
Catalog-Aware Kerberos Authentication
================================== [docutils]
```

> Catalog-Aware Kerberos Authentication

Reviewer: The file name is ``hive-catalog-aware-kerberos.rst``, so consider revising the page title to include Hive. Maybe `Hive Catalog-Aware Kerberos Authentication`, `Catalog-Aware Kerberos Authentication for Hive`, or something you feel accurately describes the page. 

Note: If you change this line, please make sure the formatting lines in lines 1 and 3 match the length of line 2.

Author: All issues should now be addressed. I really appreciate your guidance.

> ==================================

Reviewer: ```suggestion
=====================================
```

Overview
--------

Catalog-aware Kerberos authentication allows Presto to use different Kerberos authentication rules for different Hive catalogs. This is particularly useful in multi-domain environments where different Hive clusters may belong to different Kerberos realms and require different ``auth_to_local`` rules for principal mapping.

Traditional Kerberos authentication in Presto uses a global configuration that applies to all catalogs, which can cause issues when working with multiple Kerberos domains. The catalog-aware approach solves this by maintaining separate authentication contexts for each catalog.

Configuration
-------------

> To enable catalog-aware Kerberos authentication, you need to configure the following properties in your Hive catalog configuration files.

Reviewer: ```suggestion
To enable catalog-aware Kerberos authentication, you must configure the following properties in your Hive catalog configuration files.
```

Basic Configuration
^^^^^^^^^^^^^^^^^^^

> In your catalog properties file (e.g., ``etc/catalog/hive.properties``):

Reviewer: ```suggestion
In your catalog properties file (for example, ``etc/catalog/hive.properties``):
```
Avoid Latin abbreviations. See the [GitLab documentation style guide recommended word list entry for e.g.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg) for discussion and other suggestions. 

I've made one suggestion here, please revise my suggestion if you like an alternative better.
